### PR TITLE
Z5 map

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
[BALANCE][TRIVIAL][ENHANCEMENT]
## About the PR 
- Makes it so the Z5 level is also mapped on Space maps.
- Adds the map to PDAs on Space maps.
- Fixes a runtime when examining the trench map item.
- Makes it so hotspots are only generated when playing an Ocean map.

image

Also, I apparently accidentally mixed some of tarm's upcoming smoke patch in. *scream
But I reverted it, so my changes should be fine now.
Sorry for the inconvenience, I can't into git.
## Why's this needed? 

- It'll make exploring Z5 for structures less annoying.

## Changelog

```
(u)zjdtmkhzt:
(*)The mining z-level now has a map. Find it in your PDA!
```

### CODE REVIEW

**Comment by Gannets** [comment]
A much needed feature.

File: code/datums/controllers/sea_hotspot_controls.dm [hunk 1]
```diff
@@ -25,31 +26,38 @@
...
 			map = icon('icons/misc/trenchMapEmpty.dmi', "template")
 			var/turf_color = null
 			for (var/x = 1, x <= world.maxx, x++)
 				for (var/y = 1, y <= world.maxy, y++)
 					var/turf/T = locate(x,y,5)
-
 					if (T.name == "asteroid" || T.name == "cavern wall" || T.type == /turf/simulated/floor/plating/airless/asteroid)
 						turf_color = "solid"
-					else if (T.name == "trench floor")
+					else if (T.name == "trench floor" || T.name == "\proper space")
```
Comment (pali6) [inline code/datums/controllers/sea_hotspot_controls.dm:60]: Hm, I wonder why all those checks are name based ant not type based.

**Comment by pali6** [comment]
Very good PR. I promised I'd make this at least a 5 times and never got around to doing it. 😅 And I'd probably overengineer it anyway. Thanks.

**Comment by Carbadox** [comment]
actually really cool and awesome. Nice stuff

**Pull request merged**